src/ui/public/test_harness/test_sharding/setup_test_sharding.js

```diff
@@ -0,0 +1,50 @@
+import { uniq, defaults } from 'lodash';
+
+import { findTestBundleUrl } from './find_test_bundle_url';
+import { getShardingParamsFromUrl } from './get_sharding_params_from_url';
+import { setupRootDescribeCallFilter } from './root_describe_call_filter';
+import { getShardNum } from './get_shard_num';
+
+const DEFAULT_PARAMS = {
+  shards: 1,
+  shard_num: 1
+};
+
+export function setupTestSharding() {
+  const pageUrl = window.location.href;
+  const bundleUrl = findTestBundleUrl();
+
+  // read config from the page url, and from the
```

Comment:
Doesn't it look strange to overwrite a bundle URL's params with the page URL's params? Wouldn't someone reading this wonder why that's being done?